foreman/data_refinery_foreman/surveyor/sra.py

```diff
@@ -267,10 +273,11 @@ def gather_study_metadata(metadata: Dict) -> None:
     def gather_all_metadata(run_accession):
         metadata = SraSurveyor.gather_run_metadata(run_accession)
 
-        SraSurveyor.gather_experiment_metadata(metadata)
-        SraSurveyor.gather_sample_metadata(metadata)
-        SraSurveyor.gather_study_metadata(metadata)
-        SraSurveyor.gather_submission_metadata(metadata)
+        if metadata != {}:
```

Comment:
Again, if metadata is {} we have nothing of value to work with and this is probably a fail..

Reply:
I agree. I supposed I could have raised an exception and then caught it higher up, but instead I added some additional logic to make sure that the dictionary returned by these functions isn't empty. If you feel strongly about it can I refactor it to use exceptions.